File: src/main/java/com/sprout/beemod/BeeMod.java
package com.sprout.beemod;

import com.sprout.beemod.block.ModBlocks;
import com.sprout.beemod.block.entity.ModBlockEntities;
import com.sprout.beemod.effect.ModEffects;
import com.sprout.beemod.entity.ModEntities;
import com.sprout.beemod.item.ModItems;
import com.sprout.beemod.poitype.ModPoiTypes;
import net.minecraft.world.level.block.Blocks;
import net.minecraft.world.level.block.FlowerPotBlock;
import net.neoforged.bus.api.IEventBus;
import net.neoforged.fml.common.Mod;
import net.neoforged.fml.event.lifecycle.FMLCommonSetupEvent;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Mod(BeeMod.MOD_ID)
public class BeeMod {
    public static final String MOD_ID = "beemod";
    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);

    public BeeMod(IEventBus modEventBus) {
        LOGGER.info("Hello NeoForge world!");

        ModItems.register(modEventBus);
        ModBlocks.register(modEventBus);
        ModEffects.register(modEventBus);
        ModBlockEntities.register(modEventBus);
        ModEntities.register(modEventBus);
        ModPoiTypes.register(modEventBus);

        modEventBus.addListener(this::commonSetup);
    }

    private void commonSetup(final FMLCommonSetupEvent event) {
        event.enqueueWork(this::addFlowers);
    }

    private void addFlowers() {
        FlowerPotBlock potBlock = ((FlowerPotBlock) Blocks.FLOWER_POT);
        potBlock.addPlant(ModBlocks.SOULBLOSSOM.getId(), ModBlocks.POTTED_SOULBLOSSOM);
    }

}

File: src/main/kotlin/com/sprout/beemod/BeeMod.kt
package com.sprout.beemod

import com.mojang.logging.LogUtils
import com.sprout.beemod.datagen.DataGenerators
import com.sprout.beemod.item.ModItems
import net.neoforged.bus.api.IEventBus
import net.neoforged.fml.common.Mod
import net.neoforged.fml.event.lifecycle.FMLCommonSetupEvent
import net.neoforged.neoforge.common.NeoForge
import org.slf4j.Logger

@Mod(BeeMod.MODID)
class BeeMod(modEventBus: IEventBus) {
    init {
        // Register the commonSetup method for modloading
        modEventBus.addListener { _: FMLCommonSetupEvent -> this.commonSetup() }

        ModItems.register(modEventBus)
    }

    private fun commonSetup() {
        // Some common setup code
        LOGGER.info("HELLO FROM COMMON SETUP")
    }

    companion object {
        // Define mod id in a common place for everything to reference
        const val MODID: String = "beemod"

        // Directly reference a slf4j logger
        val LOGGER: Logger = LogUtils.getLogger()
    }
}
